build-aux/snap/snapcraft.yaml

```diff
@@ -50,9 +53,48 @@ parts:
       # install build dependencies
       export DEBIAN_FRONTEND=noninteractive
       export DEBCONF_NONINTERACTIVE_SEEN=true
-      sudo -E apt-get build-dep -y ./
+      sudo -E apt-get build-dep -y "$SNAPCRAFT_PART_SRC"
+      # Get archives from xenial; we are not shipping these, we just need to
+      # build against them
+      if [ "${SNAPCRAFT_TARGET_ARCH}" = "amd64" ] || [ "${SNAPCRAFT_TARGET_ARCH}" = "i386" ]; then
+          UBUNTU_POOL="archive.ubuntu.com/ubuntu/pool"
+      else
+          UBUNTU_POOL="ports.ubuntu.com/pool"
+      fi
```

Comment:
will this get us the ESM bits for amd64 when needed?

Reply:
It won't but that is okay, we only but it here but never actually ship this version, there is a separate `libc6` step in the snapcraft.yaml that will get us the core20 libc6. This libc6 here is only used to resolve symbols so that go does not use newer glibc symbols. Maybe I should expand the comment? Or am I misunderstanding something about the question ?

Reply:
I see, I think the comment needs a bit of expanding, probably also mentioned the step that actually provided the libc6 we actually ship

Reply:
I expanded the comment